typescript/sdk/src/router.ts

```diff
@@ -1,20 +1,22 @@
-import { Router, Router__factory } from '@abacus-network/app';
-import {
-  AbacusConnectionManager,
-  AbacusConnectionManager__factory,
-} from '@abacus-network/core';
+import { ethers } from 'ethers';
+
+import { Router } from '@abacus-network/app';
 
 import { AbacusContracts, AbacusFactories } from './contracts';
 
 export type RouterContracts<RouterContract extends Router = Router> =
   AbacusContracts & {
     router: RouterContract;
-    abacusConnectionManager: AbacusConnectionManager;
   };
 
-export type RouterFactories<
-  RouterFactory extends Router__factory = Router__factory,
-> = AbacusFactories & {
-  router: RouterFactory;
-  abacusConnectionManager: AbacusConnectionManager__factory;
-};
+type RouterFactory<RouterContract extends Router = Router> =
+  ethers.ContractFactory & {
+    deploy: (...args: any[]) => Promise<RouterContract>;
+  };
+
+export type RouterFactories<RouterContract extends Router = Router> =
+  AbacusFactories & {
+    router: RouterFactory<RouterContract>;
+  };
+
+export { Router } from '@abacus-network/app';
```

Comment:
Why the re-export here? Should this stuff be moved to `@abacus-network/app` with the Router class itself?

Reply:
oh, interesting
maybe `app` should be `router` and contain the contracts, deploy, and app router stuff